changelog/dmd.two-pass-inliner.dd

```diff
@@ -0,0 +1,25 @@
+The compiler now inlines `pragma(inline, true)` functions in a separate pass
+
+The compiler now considers `pragma(inline, true)` functions for inlining in a dedicated pass
+before trying to inline other eligible functions. This provides better control of inlining
+decisions, for example:
+
+```d
+auto staticSquares(uint n)()
+{
+    int[n] arr;
+    static foreach(i; 0 .. n)
+        arr[i] = (i + 1) ^^ 2;
+    return arr;
+}
+
+pragma(inline, true) int thirtyThirty()
+{
+    return staticSquares!30[$ - 1];
+}
+```
+
+Previously, `dmd -inline` would first inline `staticSquares()` into `thirtyThirty()` and
```

Comment:
Thanks for the write-up for the changelog. My impression was that there are other issues but the size of the code that prevent inlining in one pass. If we remove the restriction on code size for forced inlining, would the current one-pass inliner also work consistently?

Reply:
Ah, thanks for the info. No problem, can stay as is.